test/inductor/test_compiled_optimizers.py

```diff
@@ -13,12 +14,95 @@
 # The rest of the optimizers not yet imported: Adamax, LBFGS, RAdam, SGD, SparseAdam
 from torch.optim import Adadelta, Adagrad, Adam, AdamW, ASGD, NAdam, RMSprop, Rprop
 
+from torch.testing._internal.common_optimizers import optim_db
+
 from torch.testing._internal.common_utils import TestCase
 
 from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
 
+
+class KernelCounts(NamedTuple):
+    multitensor: int
+    singletensor: int
+
+
+# With different settings for certain
+# tests you can get different kernel counts
+# This maps the test name to the
+# expected kernel count
+KERNEL_COUNT_OVERRIDES = {
+    "test_rmsprop_foreach_cpu_weight_decay": 12,
+    "test_nadam_foreach_cpu_weight_decay_momentum_decay": 20,
+    "test_adamw_foreach_cuda_amsgrad_capturable": 3,
+    "test_adamw_cuda_amsgrad_capturable": 6,
+    "test_adam_cuda_amsgrad_capturable": 6,
+    "test_adadelta_foreach_cpu_weight_decay_maximize": 12,
+    "test_adadelta_foreach_cpu_rho_weight_decay": 12,
+    "test_adadelta_foreach_cpu_weight_decay": 12,
+}
+
+# also tracks currently supported optimizers
+KERNEL_COUNTS = {
+    Adam: KernelCounts(multitensor=2, singletensor=8),
+    AdamW: KernelCounts(multitensor=2, singletensor=8),
+    NAdam: KernelCounts(multitensor=2, singletensor=12),
+    Rprop: KernelCounts(multitensor=1, singletensor=4),
+    RMSprop: KernelCounts(multitensor=1, singletensor=4),
+    Adadelta: KernelCounts(multitensor=1, singletensor=4),
+    Adagrad: KernelCounts(multitensor=5, singletensor=8),
+    ASGD: KernelCounts(multitensor=2, singletensor=12),
+}
+
+
+def build_compiled_opt_kwarg_db():
+    compiled_opt_db = []
+    for optim_info in optim_db:
+        if optim_info.optim_cls not in KERNEL_COUNTS:
+            continue
+
+        for optim_inputs in optim_info.optim_inputs_func():
+            for device in ["cpu", "cuda:0"]:
```

Comment:
```suggestion
            for device in ["cpu", "cuda"]:
```
why not just cuda here?